api-operator/pkg/endpoints/sidecar.go

```diff
@@ -38,7 +39,7 @@ const (
 
 var logger = log.Log.WithName("endpoints.sidecar")
 
-func AddSidecarContainers(client *client.Client, apiNamespace string, sidecarEpNames *map[string]bool) error {
+func AddSidecarContainers(client *client.Client, apiNamespace string, sidecarEpNames *map[string]bool, containersList *[]corev1.Container) error {
 	containerList := make([]corev1.Container, 0, len(*sidecarEpNames))
```

Comment:
Shall we remove this `containerList` variable  and append sidecar containers directly.